lib/heap/heap.c

```diff
@@ -387,12 +387,13 @@ static bool inplace_realloc(struct sys_heap *heap, void *ptr, size_t bytes)
 {
 	struct z_heap *h = heap->heap;
 
-	if (size_too_big(h, bytes)) {
+	chunkid_t c = mem_to_chunkid(h, ptr);
+	size_t align_gap = (uint8_t *)ptr - (uint8_t *)chunk_mem(h, c);
```

Comment:
This bit doesn't look right.  That alignment gap reflects the space needed above the original block's start.  We're moving to a different area now with arbitrary chunk alignment, and the gap will be different.  There's no single "size that is too big", but I guess you could make a conservative guess by using the value from a pessimal alignment.